[dlib] Updated dlib to v19.19

stoperro opened the pull request:
Updated dlib from v19.17 to v19.19 which at the time of writing is newest official release: Dec 14, 2019 (http://dlib.net/release_notes.html)

Additional changes to port files I had to do:
- As I did minimal changes to original dlib files, now dlib may test if libpng and libjpeg works. Still port files are patching original CMake files, so that libpng and libjpeg are REQUIRED and hopefully will fail if not found/working.
- Removed patch for Mac as it failed to apply on upstream, and looked as no longer necessary (original CMakeLists.txt no longer checks for APPLE there).

So far tested only on x64-windows, namely dlib[cuda]:x64-windows.

stoperro committed:
[dlib] Updated dlib to v19.19

msftclas commented:
[![CLA assistant check](https://cla.opensource.microsoft.com/pull/badge/not_signed)](https://cla.opensource.microsoft.com/microsoft/vcpkg?pullRequest=10826) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/vcpkg?pullRequest=10826) before we can accept your contribution.<br/><br/>:x: stoperro [sign now](https://cla.opensource.microsoft.com/microsoft/vcpkg?pullRequest=10826)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.opensource.microsoft.com/check/microsoft/vcpkg?pullRequest=10826) it.</sub>

LilyWangL commented on ports/dlib/portfile.cmake:
```diff
@@ -5,11 +5,10 @@ vcpkg_check_linkage(ONLY_STATIC_LIBRARY)
 vcpkg_from_github(
```
`include(vcpkg_common_functions)` is deprecated. Please remove it.

LilyWangL commented on ports/dlib/CONTROL:
```diff
@@ -1,5 +1,5 @@
 Source: dlib
-Version: 19.17-1
+Version: 19.19-1
```
This version should be 19.19.

LilyWangL commented on ports/dlib/portfile.cmake:
```diff
@@ -67,6 +66,8 @@ file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/include/dlib/cmake_utils/test_for_cu
+file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/include/dlib/cmake_utils/test_for_libpng)
 file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/include/dlib/external/libpng/arm)
 
 # Dlib encodes debug/release in its config.h. Patch it to respond to the NDEBUG macro instead.
```
Can you please use command `file(INSTALL ${SOURCE_PATH}/dlib/LICENSE.txt DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)` to handle copyright?

stoperro committed:
- [dlib] Corrected dlib version.
- [dlib] Corrected copyright installation.
- [dlib] Removed deprecated calls from portfile.

stoperro commented:
Hmm, the CI fail is in "osg" port, which I suspect doesn't depend on dlib. Is it some flacky test situation?

ras0219-msft commented:
LGTM, thanks for the update!
